src/Worker/Core/Shims/TaskEntityShim.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DurableTask.Core;
+using DurableTask.Core.Entities;
+using DurableTask.Core.Entities.OperationFormat;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+using DTCore = DurableTask.Core;
+
+namespace Microsoft.DurableTask.Worker.Shims;
+
+/// <summary>
+/// Shim that provides the entity context and implements batched execution.
+/// </summary>
+class TaskEntityShim : DTCore.Entities.TaskEntity
+{
+    readonly DataConverter dataConverter;
+    readonly ITaskEntity taskEntity;
+    readonly EntityInstanceId entityId;
+
+    readonly StateShim state;
+    readonly ContextShim context;
+    readonly OperationShim operation;
+
+    // entities roll back the state and actions when the user code throws an unhandled exception
+    string? checkpointedState;
+    int checkpointedPosition;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TaskEntityShim"/> class.
+    /// </summary>
+    /// <param name="dataConverter">The data converter.</param>
+    /// <param name="taskEntity">The task entity.</param>
+    /// <param name="entityId">The entity ID.</param>
+    public TaskEntityShim(
+        DataConverter dataConverter,
+        ITaskEntity taskEntity,
+        EntityId entityId)
+    {
+        this.dataConverter = Check.NotNull(dataConverter);
+        this.taskEntity = Check.NotNull(taskEntity);
+        this.entityId = new EntityInstanceId(entityId.Name, entityId.Key);
+        this.state = new StateShim(this, dataConverter);
+        this.context = new ContextShim(this, dataConverter);
+        this.operation = new OperationShim(this);
+    }
+
+    /// <inheritdoc />
+    public override async Task<EntityBatchResult> ExecuteOperationBatchAsync(EntityBatchRequest operations)
+    {
+        // initialize/reset the state and action list
+        this.state.CurrentState = operations.EntityState;
+        this.context.Rollback(0);
```

Comment:
If you go with the commit/rollback suggestion of mine, can make this a `Reset()`.

Reply:
I have revised the Commit/Rollback/Reset.